test/jdk/java/lang/module/ModuleDescriptorHashCodeTest.java

```diff
@@ -63,6 +62,45 @@ public void testBootModuleDescriptor() throws Exception {
         }
     }
 
+    /**
+     * Verifies that two "equal" module descriptors which only differ in the order of
+     * {@link ModuleDescriptor.Opens.Modifier opens modifiers}, that were used to construct the
+     * descriptors, have the same hashcode.
+     */
+    @Test
+    public void testModifiersOrdering() throws Exception {
+        // create a module descriptor
+
+        // important to use Set.of() (i.e. backed by immutable set) to reproduce the issue
+        final Set<ModuleDescriptor.Opens.Modifier> mod1 = Set.of(
+                ModuleDescriptor.Opens.Modifier.SYNTHETIC, ModuleDescriptor.Opens.Modifier.MANDATED);
```

Comment:
Thieseshould probably be "mods1" and "mods2" rather than "mod1" and "mod2" as they are a set of modifiers rather than a single modifier.

Reply:
Fixed in the latest version of this PR.